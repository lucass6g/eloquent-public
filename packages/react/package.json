{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
