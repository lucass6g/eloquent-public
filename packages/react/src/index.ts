export {Switch, Case, Default} from './components/conditional-switch'