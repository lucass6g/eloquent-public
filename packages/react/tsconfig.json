{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
