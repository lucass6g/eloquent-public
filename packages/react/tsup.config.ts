import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})