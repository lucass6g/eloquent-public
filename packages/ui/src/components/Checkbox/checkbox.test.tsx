import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { Checkbox } from '.';
describe('Checkbox Component', () => {
    it('renders the default Checkbox', () => {
        const component =  render(
            <Checkbox.Root>
                <Checkbox.Button id="terms" />
                <Checkbox.Label htmlFor="terms">Aceitar termos e condições da promoção</Checkbox.Label>
                <Checkbox.HelperText>Você concorda com nossos Termos de Serviço e Política de Privacidade.</Checkbox.HelperText>
            </Checkbox.Root>
        );
        const checkbox = screen.getByRole('checkbox', { name: /aceitar termos e condições da promoção/i });
        const helperText = screen.getByText(/você concorda com nossos termos de serviço e política de privacidade/i);
        expect(checkbox).toBeInTheDocument();
        expect(helperText).toBeInTheDocument();
        expect(checkbox).not.toBeChecked();
        fireEvent.click(checkbox);
        expect(checkbox).toBeChecked();
        expect(component.baseElement).toMatchSnapshot();
    });
    it('renders the disabled Checkbox', () => {
        const component = render(
            <Checkbox.Root disabled>
                <Checkbox.Button disabled id="terms" />
                <Checkbox.Label htmlFor="terms">Aceitar termos e condições da promoção</Checkbox.Label>
                <Checkbox.HelperText>Você concorda com nossos Termos de Serviço e Política de Privacidade.</Checkbox.HelperText>
            </Checkbox.Root>
        );
        const checkbox = screen.getByRole('checkbox', { name: /aceitar termos e condições da promoção/i });
        expect(checkbox).toBeDisabled();
        expect(component.baseElement).toMatchSnapshot();
    });
    it('renders Checkbox with label', () => {
        render(
            <Checkbox.Root>
                <Checkbox.Button id="terms1" />
                <Checkbox.Label htmlFor="terms1">Aceitar termos e condições da promoção</Checkbox.Label>
            </Checkbox.Root>
        );
        const checkbox = screen.getByRole('checkbox', { name: /aceitar termos e condições da promoção/i });
        expect(checkbox).toBeInTheDocument();
        expect(checkbox).not.toBeChecked();
        fireEvent.click(checkbox);
        expect(checkbox).toBeChecked();
    });
    it('renders Checkbox with helper text', () => {
        render(
            <Checkbox.Root>
                <Checkbox.Button id="terms2" />
                <Checkbox.Label htmlFor="terms2">Aceitar termos e condições da promoção</Checkbox.Label>
                <Checkbox.HelperText>Você concorda com nossos Termos de Serviço e Política de Privacidade.</Checkbox.HelperText>
            </Checkbox.Root>
        );
        const helperText = screen.getByText(/você concorda com nossos termos de serviço e política de privacidade/i);
        expect(helperText).toBeInTheDocument();
    });
});