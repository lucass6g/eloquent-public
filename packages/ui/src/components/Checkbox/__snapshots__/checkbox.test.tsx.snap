// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox Component > renders the default Checkbox 1`] = `
<body>
  <div>
    <div
      class="flex-col space-x-2 [&[aria-disabled='true']]:cursor-not-allowed [&[aria-disabled='true']]:opacity-50"
    >
      <button
        aria-checked="true"
        class="peer h-4 w-4 shrink-0 rounded-[4px] border border-primary shadow focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
        data-state="checked"
        id="terms"
        role="checkbox"
        type="button"
        value="on"
      >
        <span
          class="flex items-center justify-center text-current"
          data-state="checked"
          style="pointer-events: none;"
        >
          <svg
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.4669 3.72684C11.7558 3.91574 11.8369 4.30308 11.648 4.59198L7.39799 11.092C7.29783 11.2452 7.13556 11.3467 6.95402 11.3699C6.77247 11.3931 6.58989 11.3355 6.45446 11.2124L3.70446 8.71241C3.44905 8.48022 3.43023 8.08494 3.66242 7.82953C3.89461 7.57412 4.28989 7.55529 4.5453 7.78749L6.75292 9.79441L10.6018 3.90792C10.7907 3.61902 11.178 3.53795 11.4669 3.72684Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
      <label
        class="text-sm font-medium text-green-dark-600"
        for="terms"
      >
        Aceitar termos e condições da promoção
      </label>
      <span
        class="block pl-4 text-sm text-neutral-500"
      >
        Você concorda com nossos Termos de Serviço e Política de Privacidade.
      </span>
    </div>
  </div>
</body>
`;

exports[`Checkbox Component > renders the disabled Checkbox 1`] = `
<body>
  <div>
    <div
      aria-disabled="true"
      class="flex-col space-x-2 [&[aria-disabled='true']]:cursor-not-allowed [&[aria-disabled='true']]:opacity-50"
    >
      <button
        aria-checked="false"
        class="peer h-4 w-4 shrink-0 rounded-[4px] border border-primary shadow focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
        data-disabled=""
        data-state="unchecked"
        disabled=""
        id="terms"
        role="checkbox"
        type="button"
        value="on"
      />
      <label
        class="text-sm font-medium text-green-dark-600"
        for="terms"
      >
        Aceitar termos e condições da promoção
      </label>
      <span
        class="block pl-4 text-sm text-neutral-500"
      >
        Você concorda com nossos Termos de Serviço e Política de Privacidade.
      </span>
    </div>
  </div>
</body>
`;
